geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,411 @@
+
+        :return: The value store in the metadata if exists.
+        """
+        metadata = self.metadata if self.metadata is not None else {}
+        return metadata.get(key, None)
+
+    def _update_to_metadata(self, key: str, values: list | None):
+        """
+        Update the metadata with the interpretation curves.
+
+        :param key: The key to update.
+        :param values: The values to update.
+        """
+        if values:
+            self.update_metadata(
+                {
+                    key: [
+                        value.map_back
+                        if isinstance(value, InterpretationSectionParams)
+                        else str(value.uid)
+                        for value in values
+                    ]
+                }
+            )
+        else:
+            setattr(self, key, None)
+            self.update_metadata({key: None})
+
+    def _verify_object(self, value, key: str):
+        """
+        Get the object and verify that it is a children of the group.
+
+        :param value: The value to verify.
+        :param key: The default name of the expected object type.
+
+        :return: The verified object
+        """
+        if isinstance(value, str | UUID):
+            value = str2uuid(value)
+            value = self.workspace.get_entity(value)[0]
+
+        if getattr(value, "_default_name", None) != key:
+            raise TypeError(
+                f"The {value} object must be a {key} object."
+                f" {type(value)} provided."
+            )
+
+        return value
+
+    def add_interpretation_curve(self, curves: Curve | list[Curve]):
+        """
+        Add an interpretation curve.
+
+        :param curves: The interpretation curve to add.
+        """
+        if not isinstance(curves, list):
+            curves = [curves]
+
+        self._load_attributes()
+        for curve in curves:
+            curve = self._verify_object(curve, "Curve")
+            self.add_children(curve)
+            if self._interpretation_curves is None:
+                self._interpretation_curves = [curve]
+            elif curve not in self._interpretation_curves:
+                self._interpretation_curves.append(curve)
+
+        self._update_to_metadata("Interpretation curves", self._interpretation_curves)
+
+    def add_interpretation_section(
+        self, sections: InterpretationSectionParams | list[InterpretationSectionParams]
+    ):
+        """
+        Add an interpretation section.
+
+        :param sections: The interpretation section to add.
+        """
+        if not isinstance(sections, list):
+            sections = [sections]
+
+        self._load_attributes()
+        for section in sections:
+            section = InterpretationSectionParams.verify(section)
+            if self._interpretation_sections is None:
+                self._interpretation_sections = [section]
+            elif section.in_list(self._interpretation_sections) is None:
```

Comment:
Flip the logic here.
```suggestion
            elif section not in self._interpretation_sections:
```